src/main/java/com/kruize/analysis/runtimes/java/OpenJ9AnalysisImpl.java

```diff
@@ -0,0 +1,96 @@
+
+import com.kruize.metrics.MetricsImpl;
+import com.kruize.metrics.runtimes.java.JavaApplicationMetricsImpl;
+import com.kruize.metrics.runtimes.java.JavaMetricCollector;
+import com.kruize.metrics.runtimes.java.openj9.OpenJ9MetricCollector;
+import com.kruize.util.MathUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class OpenJ9AnalysisImpl
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(OpenJ9AnalysisImpl.class);
+
+    public static void analyseHeapRecommendation(MetricsImpl metrics)
+    {
+        for (String application : JavaApplicationMetricsImpl.javaApplicationMetricsMap.keySet()) {
+            double maxHeap = 0;
+
+            for (JavaMetricCollector metricCollector :
+                    JavaApplicationMetricsImpl.javaApplicationMetricsMap.get(application))
+            {
+                OpenJ9MetricCollector openJ9MetricCollector = (OpenJ9MetricCollector) metricCollector;
+                maxHeap = (maxHeap > openJ9MetricCollector.getHeap()) ? maxHeap : openJ9MetricCollector.getHeap();
+            }
+
+            double numberOfMetrics = JavaApplicationMetricsImpl.javaApplicationMetricsMap
+                    .get(application).size();
+            double heapRecommendation = maxHeap;
+
+            if (MathUtil.bytesToMB(heapRecommendation) > metrics.getRssRequests())
+            {
+                double MbToBytes = 1000*1000;
+                heapRecommendation = 0.7 * metrics.getRssRequests() * MbToBytes;
+            }
+
+            JavaApplicationMetricsImpl.javaApplicationInfoMap
+                    .get(application)
+                    .getJavaRecommendations()
+                    .setHeapRecommendation(heapRecommendation);
+
+            LOGGER.info("Heap recommendation for {} is {}MB", application,
+                    MathUtil.bytesToMB(heapRecommendation));
+
+        }
+    }
+
+    public static void analyseNonHeapRecommendation(MetricsImpl metrics)
+    {
+        for (String application : JavaApplicationMetricsImpl.javaApplicationMetricsMap.keySet()) {
+            double maxNonHeap = 0;
+
+            for (JavaMetricCollector metricCollector :
+                    JavaApplicationMetricsImpl.javaApplicationMetricsMap.get(application))
+            {
+                OpenJ9MetricCollector openJ9MetricCollector = (OpenJ9MetricCollector) metricCollector;
+                maxNonHeap = (maxNonHeap > openJ9MetricCollector.getNonHeap()) ? maxNonHeap : openJ9MetricCollector.getNonHeap();
+            }
+
+            double numberOfMetrics = JavaApplicationMetricsImpl.javaApplicationMetricsMap
+                    .get(application).size();
+            double nonHeapRecommendation = maxNonHeap;
+
+            if (MathUtil.bytesToMB(nonHeapRecommendation) > metrics.getRssRequests())
+            {
+                double MbToBytes = 1000*1000;
+                nonHeapRecommendation = 0.7 * metrics.getRssRequests() * MbToBytes;
```

Comment:
Would be good to add some documentation for each of the functions explaining the logic and the rationale.

Reply:
Added documentation